internal/distro/rhel7/distro.go

```diff
@@ -0,0 +1,542 @@
+	}
+	return bootType
+}
+
+func (t *imageType) getPartitionTable(
+	mountpoints []blueprint.FilesystemCustomization,
+	options distro.ImageOptions,
+	rng *rand.Rand,
+) (*disk.PartitionTable, error) {
+	archName := t.arch.Name()
+
+	basePartitionTable, exists := t.basePartitionTables[archName]
+
+	if !exists {
+		return nil, fmt.Errorf("unknown arch: " + archName)
+	}
+
+	imageSize := t.Size(options.Size)
+
+	return disk.NewPartitionTable(&basePartitionTable, mountpoints, imageSize, true, rng)
+}
+
+func (t *imageType) getDefaultImageConfig() *distro.ImageConfig {
+	// ensure that image always returns non-nil default config
+	imageConfig := t.defaultImageConfig
+	if imageConfig == nil {
+		imageConfig = &distro.ImageConfig{}
+	}
+	return imageConfig.InheritFrom(t.arch.distro.getDefaultImageConfig())
+
+}
+
+func (t *imageType) PartitionType() string {
+	archName := t.arch.Name()
+	basePartitionTable, exists := t.basePartitionTables[archName]
+	if !exists {
+		return ""
+	}
+
+	return basePartitionTable.Type
+}
+
+func (t *imageType) Manifest(customizations *blueprint.Customizations,
+	options distro.ImageOptions,
+	repos []rpmmd.RepoConfig,
+	packageSpecSets map[string][]rpmmd.PackageSpec,
+	seed int64) (distro.Manifest, error) {
+
+	if err := t.checkOptions(customizations, options); err != nil {
+		return distro.Manifest{}, err
+	}
+
+	source := rand.NewSource(seed)
+	// math/rand is good enough in this case
+	/* #nosec G404 */
+	rng := rand.New(source)
+
+	pipelines, err := t.pipelines(t, customizations, options, repos, packageSpecSets, rng)
+	if err != nil {
+		return distro.Manifest{}, err
+	}
+
+	// flatten spec sets for sources
+	allPackageSpecs := make([]rpmmd.PackageSpec, 0)
+	for _, specs := range packageSpecSets {
+		allPackageSpecs = append(allPackageSpecs, specs...)
+	}
+
+	// handle inline sources
+	inlineData := []string{}
+
+	// FDO root certs, if any, are transmitted via an inline source
+	if fdo := customizations.GetFDO(); fdo != nil && fdo.DiunPubKeyRootCerts != "" {
+		inlineData = append(inlineData, fdo.DiunPubKeyRootCerts)
+	}
+
+	return json.Marshal(
+		osbuild.Manifest{
+			Version:   "2",
+			Pipelines: pipelines,
+			Sources:   t.sources(allPackageSpecs, inlineData),
+		},
+	)
+}
+
+func (t *imageType) sources(packages []rpmmd.PackageSpec, inlineData []string) osbuild.Sources {
+	sources := osbuild.Sources{}
+	curl := &osbuild.CurlSource{
+		Items: make(map[string]osbuild.CurlSourceItem),
+	}
+	for _, pkg := range packages {
+		item := new(osbuild.URLWithSecrets)
+		item.URL = pkg.RemoteLocation
+		if pkg.Secrets == "org.osbuild.rhsm" {
+			item.Secrets = &osbuild.URLSecrets{
+				Name: "org.osbuild.rhsm",
+			}
+		}
+		curl.Items[pkg.Checksum] = item
+	}
+	if len(curl.Items) > 0 {
+		sources["org.osbuild.curl"] = curl
+	}
+
+	if len(inlineData) > 0 {
+		ils := osbuild.NewInlineSource()
+		for _, data := range inlineData {
+			ils.AddItem(data)
+		}
+
+		sources["org.osbuild.inline"] = ils
+	}
+
+	return sources
+}
```

Comment:
This should be removed in favor of `osbuild2.GenSources()`

Reply:
Ah, cool, yes! Done!